### PR TITLE
(#3279) Autosuggest Setup

### LINKED ISSUE #3279: Build: Header Search Box Autosuggest Wireup JS
The search box in the header is a text input and submit button that submits to a search results page. This ticket is to enhance that text input as a NCIDS autosuggest component hitting the [Sitewide Search Autosuggest API Endpoint](https://webapis.cancer.gov/sitewidesearch/v1/index.html#/Autosuggest).

This endpoint requires the following parameters along with the source of that information:
* Collection - *required* - this should be a data-attribute, `data-autosuggest-collection` on the search box input element. 
  - the absence of the collection should act as a signal to NOT initialize autocomplete
* Language - This should be retrieved from the page language. 
* Term - the text value of the input

On page load the javascript should find the input element and call Autocomplete.create() on that element. You will provide a function that will be used to fetch the data from the Autosuggest API in the format required by Autocomplete. You will also need to register to any Autocomplete events to handle analytics tracking.

<html>
<body>

English | Spanish
-- | --
Please enter more characters | Ingrese más caracteres
Please enter 3 or more characters | Ingrese 3 o más caracteres

</body>
</html>

**ANALYTICS REQUIREMENTS HERE**
``` gherkin
Scenario: Search submit
Given a user is on any page of cancer.gov
When the user searches for a term at the top of the page
Then there should be an NCIDataLayer Other event 
  And the event has the name “HeaderSearch:Submit”
  And the event has the the linkname “HeaderSearch:Submit”
  And this event should have the following data:
      | key               | value                       |
      | formType          | SitewideSearch              |
      | searchTerm        | The term entered for search |
      | autoSuggestUsage  | Selected, Offered, or None  |
      | location          | Header                      |
```

``` gherkin
Scenario: Search submit when selecting autosuggest option
Given a user is on any page of cancer.gov
When the user searches for a term at the top of the page
  And the user selected an autosuggest option
Then there should be an NCIDataLayer Other event 
  And the event has the name “HeaderSearch:Submit”
  And the event has the the linkname “HeaderSearch:Submit”
  And this event should have the following data:
      | key                 | value                                                                            |
      | formType            | SitewideSearch                                                    |
      | searchTerm          | The term entered for search                                       |
      | autoSuggestUsage    | Selected, Offered, or None                                        |
      | charactersTyped     | What the user typed in before selecting an autosuggest option     |
      |                     | Undefined if no option is selected                                |
      | numCharacters       | Number of characters typed before selecting an autosuggest option |
      |                     | Undefined if no option is selected                                |
      | numSuggestsSelected | Index of the autosuggest option clicked on                        |
      |                     | Undefined if no option is selected                                |
      | suggestItems        | Number of autosuggest items shown                                 |
      | location            | Header                                                            |
```

### PR DESCRIPTION
**Pull request by dev-rana-publicis**
ODE: https://ncigovcdode513.prod.acquia-sites.com/
Closes #3279

### CODE REVIEW

File: docroot/profiles/custom/cgov_site/themes/custom/ncids_trans/front-end/src/lib/components/nci-header/cgdp-autocomplete/cgdp-autocomplete-adapter.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+	 * @param {Axios} client an axios client.
+	 */
+	constructor(client: Axios, collection: string, lang: string) {
+		this.client = client;
+		this.collection = collection;
+		this.lang = lang;
+	}
+
+	private handleResponse(response: JSON): Array<string> {
+		const results = JSON.parse(JSON.stringify(response)).results;
```
[the remark this answers is not in the document]
Comment (bryanpizzillo) [reply docroot/profiles/custom/cgov_site/themes/custom/ncids_trans/front-end/src/lib/components/nci-header/cgdp-autocomplete/cgdp-autocomplete-adapter.ts:23]: It seems like this was just copied from the example site that does *not* use Axios. So this really should be cleaned up.

**Review by bryanpizzillo — CHANGES_REQUESTED** [review]
This also needs a correct search icon in `docroot/profiles/custom/cgov_site/themes/custom/ncids_trans/templates/block/ncids/block--ncids-header.html.twig`

**Comment by bryanpizzillo** [comment]
@dev-rana-publicis I reserve the right to have other changes, but here is a bunch.

**Review by bryanpizzillo — CHANGES_REQUESTED** [review]
Also @dev-rana-publicis can you please remove all the package stuff and move it into a separate commit preceding this one.

File: docroot/profiles/custom/cgov_site/themes/custom/ncids_trans/front-end/src/lib/components/nci-header/auto-suggest-analytics-handlers.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import { trackOther } from '../../core/analytics/eddl-util';
+
+/**
+ * Click handler for clicking on a link within the mega menu
+ * @param event
+ */
+export const searchSubmitHandler = (event: Event) => {
```
[the remark this answers is not in the document]
Comment (dev-rana-publicis) [reply docroot/profiles/custom/cgov_site/themes/custom/ncids_trans/front-end/src/lib/components/nci-header/auto-suggest-analytics-handlers.ts:7]: I am just following what was done in the mega-menu

File: docroot/profiles/custom/cgov_site/themes/custom/ncids_trans/templates/block/ncids/block--ncids-header.html.twig [hunk 3]
```diff
@@ -21,12 +21,12 @@
           {% endif %}
           <input class="usa-input" id="nci-header-search__field" type="search" name="swKeyword" {{input_attribs}} />
 
-          <button class="usa-button nci-header-search__search-button" type="submit" aria-label="{{ 'Search' | t }}">
-            <span class="nci-header-search__label" aria-hidden="true">{{ 'Search' | t }}</span>
+          <button class="usa-button nci-header-search__search-button" type="submit" aria-label="search">
```
Comment (bryanpizzillo) [inline docroot/profiles/custom/cgov_site/themes/custom/ncids_trans/templates/block/ncids/block--ncids-header.html.twig:24]: This breaks translations. You can't just copy paste here. Please fix this.

Also I thought the <svg> was the problem and I don't see that being updated.

File: docroot/profiles/custom/cgov_site/themes/custom/ncids_trans/templates/block/ncids/block--ncids-header.html.twig [hunk 4]
```diff
@@ -21,12 +21,12 @@
...
+            <span class="nci-header-search__label">Search</span>
```
Comment (bryanpizzillo) [inline docroot/profiles/custom/cgov_site/themes/custom/ncids_trans/templates/block/ncids/block--ncids-header.html.twig:25]: This breaks translations as well.

**Review by bryanpizzillo — CHANGES_REQUESTED** [review]
So I was just checking to see if the template got touched and noticed something very wrong. Please fix the two issues.

**Review by bryanpizzillo — APPROVED** [review]

**Review by bryanpizzillo — CHANGES_REQUESTED** [review]
The search button is broken on desktop, please fix.

**Review by bryanpizzillo — APPROVED** [review]

**Pull request merged**